elyra/metadata/metadata_app.py

```diff
@@ -473,6 +473,96 @@ def start(self):
               ("is" if num_invalid_exported == 1 else "are") + " invalid)")
 
 
+class SchemaspaceImport(SchemaspaceBase):
+    """Handles the 'import' subcommand functionality for a specific schemaspace."""
+
+    directory_option = CliOption("--directory", name='directory',
+                                 description='The local file system path from where the metadata will be imported',
+                                 required=True)
+
+    replace_flag = Flag("--replace", name='replace',
+                        description='Overwrite existing metadata instance with the same name',
+                        default_value=False)
+
+    # 'Import' flags
+    options: List[Option] = [directory_option, replace_flag]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.metadata_manager = MetadataManager(schemaspace=self.schemaspace)
+
+    def start(self):
+        super().start()  # process options
+
+        directory = self.directory_option.value
+
+        src_directory = os.path.join(directory, self.schemaspace)
+
+        try:
+            json_files = [f for f in os.listdir(src_directory) if f.endswith('.json')]
+        except OSError as e:
+            print(f"Unable to reach the '{src_directory}' directory: {e.strerror}: '{e.filename}'")
+            self.exit(1)
+
+        if len(json_files) == 0:
+            print(f"No instances for import found in the '{src_directory}' directory")
+            return
+
+        instance_count_imported = 0
+        instance_count_not_imported = 0
+        metadata_file = None
+
+        for file in json_files:
+            filepath = os.path.join(src_directory, file)
+            try:
+                with open(filepath) as f:
+                    metadata_file = json.loads(f.read())
+                f.close()
+            except OSError as e:
+                print(f"Unable to reach the '{file}' file: {e.strerror}: '{e.filename}'")
+                instance_count_not_imported += 1
+                continue
+
+            name = os.path.splitext(file)[0]
+            try:
+                schema_name = metadata_file['schema_name']
+                display_name = metadata_file['display_name']
+                metadata = metadata_file['metadata']
+            except KeyError as e:
+                print(f"Could not find '{e.args[0]}' key in the import file '{filepath}'")
+                instance_count_not_imported += 1
+                continue
+
+            new_instance = None
+
+            try:
+                if self.replace_flag.value:  # if replacing, fetch the instance so it can be updated
```

Comment:
I think if they've set `--replace` (or `--overwrite` per the previous comment), we still should not assume an instance exists and adjust this so it tolerates non-existence.

Reply:
I agree. This way import _optionally_ behaves like an upsert, which should be useful.